app/controllers/auth.rb

```diff
@@ -0,0 +1,18 @@
+get '/signup' do
+  erb :signup
+end
+
+post '/signup' do
+  @user = User.new(params[:user])
+  if @user.save
+    session[:user_id] = @user.id
```

Comment:
This is an idea pattern for this route except `session[:user_id] = @user.id` should be a `login_as_user(user)` helper method.